be/src/runtime/plan_fragment_executor.cpp

```diff
@@ -100,32 +113,31 @@ PlanFragmentExecutor::~PlanFragmentExecutor() {
     DCHECK(!_report_thread_active);
 }
 
-Status PlanFragmentExecutor::prepare(const TExecPlanFragmentParams& request,
-                                     QueryContext* query_ctx) {
+Status PlanFragmentExecutor::prepare(const TExecPlanFragmentParams& request) {
     OpentelemetryTracer tracer = telemetry::get_noop_tracer();
     if (opentelemetry::trace::Tracer::GetCurrentSpan()->GetContext().IsValid()) {
         tracer = telemetry::get_tracer(print_id(_query_id));
     }
     _span = tracer->StartSpan("Plan_fragment_executor");
     OpentelemetryScope scope {_span};
 
-    const TPlanFragmentExecParams& params = request.params;
-    _query_id = params.query_id;
+    if (request.__isset.query_options) {
+        _timeout_second = request.query_options.execution_timeout;
+    }
 
+    const TPlanFragmentExecParams& params = request.params;
     LOG_INFO("PlanFragmentExecutor::prepare")
             .tag("query_id", _query_id)
             .tag("instance_id", params.fragment_instance_id)
```

Comment:
fragment_instance_id will not be printed corretly, since it is a thrift type, not a pod type.